packages/headless/src/features/commerce/context/cart/cart-actions-loader.ts

```diff
@@ -5,47 +5,85 @@ import {
   CommerceEngineState,
 } from '../../../../app/commerce-engine/commerce-engine';
 import {
+  CartActionPayload,
   PurchasePayload,
   SetItemsPayload,
   UpdateItemQuantityPayload,
+  emitCartAction,
+  emitPurchase,
   purchase,
   setItems,
   updateItemQuantity,
 } from './cart-actions';
 import {cartReducer as cart} from './cart-slice';
 
-export type {PurchasePayload, SetItemsPayload, UpdateItemQuantityPayload};
+export type {
+  CartActionPayload,
+  PurchasePayload,
+  SetItemsPayload,
+  UpdateItemQuantityPayload,
+};
 
 /**
  * The cart action creators.
  */
 export interface CartActionCreators {
   /**
-   * Emits an ec_purchase analytics events with the current cart state.
+   * Emits an `ec.purchase` analytics event with the current cart state.
+   *
+   * Should be dispatched before the `purchase` action.
    *
    * @param payload - The action creator payload.
+   * @returns A dispatchable action.
    */
-  purchase(
+  emitPurchase(
     payload: PurchasePayload
   ): AsyncThunkAction<
     void,
     PurchasePayload,
     AsyncThunkCommerceOptions<CommerceEngineState>
   >;
 
-  // TODO KIT-3346: Add/expose action to emit ec_cartAction analytics events
+  /**
+   * Marks the items in the cart as purchased and empties the cart.
+   *
+   * Should be dispatched after the `emitPurchase` action.
+   *
+   * @returns A dispatchable action.
+   */
+  purchase(): PayloadAction<void>;
 
   /**
    * Sets the items in the cart.
    *
    * @param payload - The action creator payload.
+   * @returns A dispatchable action.
    */
   setItems(payload: SetItemsPayload): PayloadAction<SetItemsPayload>;
 
+  /**
+   * Emits an `ec.cartAction` analytics event.
+   *
+   * Should be dispatched before the `updateItemQuantity` action.
+   *
+   * @param payload - The action creator payload.
+   * @returns A dispatchable action.
+   */
+  emitCartAction(
```

Comment:
Similar comment as previous, should it has an `Event` prefix in that function's name?